### PR TITLE
feat: add `build_tag` env support for `set_build_info_metric`

### LINKED ISSUE #5306: Add build_tag to set_build_info_metric
Currently, we return only revision of a service in `libmetrics_build_info`:
https://github.com/neondatabase/neon/blob/ffd146c3e519ace5f14b619ca63f00ca9dda4b56/libs/metrics/src/lib.rs#L92-L100

See a grafana dashboard (with deployed revisions):
https://neonprod.grafana.net/goto/EQUiO5iSg?orgId=1

To make it more useful, let's add a build_tag to the metric, then it'll easier to find a corresponding release in https://github.com/neondatabase/neon/releases

Required changes:
- Add BUILD_TAG env var support to all services
- Add  build_tag to libmetrics_build_info

### PR DESCRIPTION
**Pull request by duguorong009**
## Problem
- Close #5306 

## Summary of changes
- Add a new util `project_build_tag` macro, similar to `project_git_version`
- Update the `set_build_info_metric` to accept and make use of `build_tag` info
- Update all codes which use the `set_build_info_metric`

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/bin/pageserver.rs [hunk 1]
```diff
@@ -262,7 +263,8 @@ fn start_pageserver(
         version(),
         launch_ts.to_string()
     );
-    set_build_info_metric(GIT_VERSION);
+    info!("build_tag: {BUILD_TAG}");
```
Comment (koivunej) [inline pageserver/src/bin/pageserver.rs:266]: Please join this with the earlier `info!` message. Rationale: we often search for the above line via `git-env:`.

Though, looking now, there's an extra `.to_string()`, and really all these three should be parameters for a message which is currently not found "Starting up".

I am not able to suggest, but something like:

```patch
-    info!(
-        "version: {} launch_timestamp: {}",
-        version(),
-        launch_ts.to_string()
-    );
-    info!("build_tag: {BUILD_TAG}");
+    info!(version=%version(), launch_timestamp=%launch_ts, build_tag=%BUILD_TAG, "Starting up");
```

File: storage_broker/src/bin/storage_broker.rs [hunk 2]
```diff
@@ -438,7 +439,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     // initialize sentry if SENTRY_DSN is provided
     let _sentry_guard = init_sentry(Some(GIT_VERSION.into()), &[]);
     info!("version: {GIT_VERSION}");
-    ::metrics::set_build_info_metric(GIT_VERSION);
+    info!("build_tag: {BUILD_TAG}");
+    ::metrics::set_build_info_metric(GIT_VERSION, BUILD_TAG);
```
Comment (koivunej) [inline storage_broker/src/bin/storage_broker.rs:443]: Does this need the `::` prefix? I cannot see from the `use` what would require it.

File: proxy/src/bin/proxy.rs [hunk 3]
```diff
@@ -96,7 +97,8 @@ async fn main() -> anyhow::Result<()> {
     let _sentry_guard = init_sentry(Some(GIT_VERSION.into()), &[]);
 
     info!("Version: {GIT_VERSION}");
-    ::metrics::set_build_info_metric(GIT_VERSION);
+    info!("Build_tag: {BUILD_TAG}");
+    ::metrics::set_build_info_metric(GIT_VERSION, BUILD_TAG);
```
Comment (koivunej) [inline proxy/src/bin/proxy.rs:101]: Same question about the `::` prefix applies than here: https://github.com/neondatabase/neon/pull/5576/files#r1368368223

File: storage_broker/src/bin/storage_broker.rs (hunk 2 above)
Comment (duguorong009) [reply storage_broker/src/bin/storage_broker.rs]: Yes, we don't need it.
Removed in https://github.com/neondatabase/neon/pull/5576/commits/bdbedec6b2c0da6e5653153e5ec55dc2404cc7dd

File: proxy/src/bin/proxy.rs (hunk 3 above)
Comment (duguorong009) [reply proxy/src/bin/proxy.rs:101]: Um, it is not possible to remove `::` prefix here.
The reason is that `::` prefix is used to indicate that `metrics` is the external crate, not internal module `metrics` of `proxy` crate.
The `proxy` crate has its own internal module `metrics` that the name is the same as external crate `metrics`.

File: pageserver/src/bin/pageserver.rs (hunk 1 above)
Comment (duguorong009) [reply pageserver/src/bin/pageserver.rs]: Done https://github.com/neondatabase/neon/pull/5576/commits/9810845dd83ee2db84541c3cb37a577b413e2ae3

**Review by conradludgate — APPROVED** [review]

File: pageserver/src/bin/pageserver.rs (hunk 1 above)
Comment (conradludgate) [reply pageserver/src/bin/pageserver.rs]: @koivunej do you feel strongly about formatted vs structured metadata here?

**Comment by bayandin** [comment]
We need to make `cargo fmt` happy:

```
+ cargo fmt --all -- --check
Diff in /__w/neon/neon/pageserver/src/bin/pageserver.rs at line 34:
 use utils::logging::TracingErrorLayerEnablement;
 use utils::signals::ShutdownSignals;
 use utils::{
-    auth::JwtAuth, logging, project_git_version, project_build_tag, sentry_init::init_sentry, signals::Signal,
-    tcp_listener,
+    auth::JwtAuth, logging, project_build_tag, project_git_version, sentry_init::init_sentry,
+    signals::Signal, tcp_listener,
 };
 
 project_git_version!(GIT_VERSION);
Diff in /__w/neon/neon/proxy/src/bin/proxy.rs at line [15](https://github.com/neondatabase/neon/actions/runs/6627941762/job/18003983542#step:9:16):
 use tokio_util::sync::CancellationToken;
 use tracing::info;
 use tracing::warn;
-use utils::{project_git_version, project_build_tag, sentry_init::init_sentry};
+use utils::{project_build_tag, project_git_version, sentry_init::init_sentry};
 
 project_git_version!(GIT_VERSION);
 project_build_tag!(BUILD_TAG);
Diff in /__w/neon/neon/safekeeper/src/bin/safekeeper.rs at line 42:
 use utils::{
     id::NodeId,
     logging::{self, LogFormat},
-    project_git_version,
-    project_build_tag,
+    project_build_tag, project_git_version,
     sentry_init::init_sentry,
     tcp_listener,
 };
Diff in /__w/neon/neon/storage_broker/src/bin/storage_broker.rs at line 44:
 };
 use utils::id::TenantTimelineId;
 use utils::logging::{self, LogFormat};
-use utils::{project_git_version, project_build_tag};
 use utils::sentry_init::init_sentry;
+use utils::{project_build_tag, project_git_version};
 
 project_git_version!(GIT_VERSION);
 project_build_tag!(BUILD_TAG);
```

https://github.com/neondatabase/neon/actions/runs/6627941762/job/18003983542

**Comment by duguorong009** [comment]
@bayandin  Thanks for tip!

Done in https://github.com/neondatabase/neon/pull/5576/commits/f716af9b13025bfc418d0da22afbadb06cd8326e

BTW, I normally run `./run_clippy.sh` for fmt and clippy check.
Maybe, is it better to add new script or update existing one so that they include this cmd `cargo fmt --all -- --check`?

File: pageserver/src/bin/pageserver.rs (hunk 1 above)
Comment (koivunej) [reply pageserver/src/bin/pageserver.rs]: For pageserver we want to use structured formatting if possible. I don't know about the other binaries the logs of which I look less.

The https://github.com/neondatabase/neon/commit/9810845dd83ee2db84541c3cb37a577b413e2ae3 is looking good enough.

**Comment by duguorong009** [comment]
> There's ./scripts/reformat which runs cargo fmt and Python formatters.
Thanks! @bayandin 
Now, I've rebased the PR with `main` branch.

**Comment by bayandin** [comment]
> Now, I've rebased the PR with main branch.

This is only required if your PR has conflicts with the main or someone asked you to do so.
Any PR changes (especially force pushed) require us to review the whole PR again (and rerunning CI again), so please refrain from doing it if it's not necessary ❤️

File: libs/metrics/src/lib.rs [hunk 4]
```diff
@@ -89,14 +89,14 @@ pub const DISK_WRITE_SECONDS_BUCKETS: &[f64] = &[
     0.000_050, 0.000_100, 0.000_500, 0.001, 0.003, 0.005, 0.01, 0.05, 0.1, 0.3, 0.5,
 ];
 
-pub fn set_build_info_metric(revision: &str) {
+pub fn set_build_info_metric(revision: &str, build_tag: &str) {
```
Comment (bayandin) [inline libs/metrics/src/lib.rs:92]: Let's add `build_tag` to a test we have: https://github.com/neondatabase/neon/blob/e0ebdfc7ce82fce9f382031aab738c6d88d3fb23/test_runner/regress/test_build_info_metric.py#L5

**Comment by duguorong009** [comment]
> > Now, I've rebased the PR with main branch.
> 
> This is only required if your PR has conflicts with the main or someone asked you to do so. Any PR changes (especially force pushed) require us to review the whole PR again (and rerunning CI again), so please refrain from doing it if it's not necessary ❤️

Oh, thanks for the tip!
I always go for rebasing the PR, conventionally.
In the future, will NOT do the rebase unless requested from reviewer or conflicts.

**Review by bayandin — APPROVED** [review]
Thanks @duguorong009 🎉

**Pull request merged**